aws_lambda_builders/registry.py

```diff
@@ -1,36 +1,99 @@
 """
-Registry of auto-discovered workflows
+Registry of auto-discovered workflows. Classes and methods in this module are thread-safe.
 """
 
-# TODO: Is this datastructure threadsafe?
-# Maintains a registry of builders
-_REGISTRY = {}
+import threading
 
+from aws_lambda_builders.exceptions import WorkflowNotFoundError
 
-def get_builder(language, language_framework, application_framework):
-    """
-    Find and return a builder class capable of building for the given combination of capabilities
 
-    :param language:
-    :param language_framework:
-    :param application_framework:
+class Registry(object):
+    """
+    Stores a registry of workflows. Only one workflow matching a given capability can be stored. If attempting to
+    register two workflows for same capability, it will raise an exception.
 
-    :return:
-    :raises BuilderNotFoundError: If a builder was not registered
+    This class is thread-safe for multiple writers (ie. registers happening simultaneously from different threads).
     """
-    pass
 
+    def __init__(self, write_lock=None):
+        self._write_lock = write_lock or threading.Lock()
+        self.data = {}
```

Comment:
A nit.  I would make this _data since I think the assumption is this is private to the class.

Reply:
Sounds good.